crates/provider/src/layers/gas.rs

```diff
@@ -122,16 +123,8 @@ where
                 tx.set_max_priority_fee_per_gas(eip1559_fees.max_priority_fee_per_gas);
                 Ok(())
             }
-            Err(err) => {
-                if err.is_transport_error()
-                    && err.to_string() == *"EIP-1559 not activated".to_string()
-                {
-                    // If EIP-1559 is not activated, it will process as a legacy tx.
-                    self.handle_legacy_tx(tx).await
-                } else {
-                    Err(err)
-                }
-            }
+            Err(RpcError::UnsupportedFeature("eip1559")) => self.handle_legacy_tx(tx).await,
```

Comment:
I think this could also fail with a
-32601, message: Method not found

rpc error for the fee_history call?

some providers also return -32600, message: Unsupported method:

so perhaps we should be less restrictive here and also check the rpc error response?

Reply:
I think that makes sense, so the unsupported feature check is okay because this looks at the base fee value of the header

Reply:
we need to write down somewhere that 1559 is somewhat privileged by the library in ways that other tx types and HF features are not. I.e. we generally assume that 1559 is active and proactively attempt to use it in gas estimation even tho we acknowledge that there may be chains that don't support it